hack/test-extended.sh

```diff
@@ -0,0 +1,101 @@
+
+export OS_EXTENDED_MASTER=127.0.0.1:${OS_EXTENDED_PORT}
+export KUBERNETES_MASTER="https://${OS_EXTENDED_MASTER}"
+
+export TMPDIR=${TMPDIR:-/tmp}
+export BASETMPDIR="${TMPDIR}/openshift-extended"
+
+# Remove all test artifacts from the previous run
+rm -rf ${BASETMPDIR} && mkdir -p ${BASETMPDIR}
+
+# Setup directories and certificates for 'curl'
+export CERT_DIR="${BASETMPDIR}/cert"
+export CURL_CA_BUNDLE="${CERT_DIR}/master/root.crt"
+export CURL_CERT="${CERT_DIR}/admin/cert.crt"
+export CURL_KEY="${CERT_DIR}/admin/key.key"
+export KUBECONFIG="${CERT_DIR}/admin/.kubeconfig"
+export OPENSHIFT_ON_PANIC=crash
+
+cleanup() {
+    set +e
+    server_pids=$(pgrep -P $(cat ${BASETMPDIR}/server.pid))
+    kill $server_pids $(cat ${BASETMPDIR}/server.pid) ${ETCD_PID}
+    rm -rf ${ETCD_DIR}
+    echo "[INFO] Cleanup complete"
+}
+
+start_server() {
+  sudo env "PATH=${PATH}" openshift start \
+    --listen="https://0.0.0.0:${OS_EXTENDED_PORT}" \
+    --public-master="https://${OS_EXTENDED_MASTER}" \
+    --etcd="http://127.0.0.1:${ETCD_PORT}" \
+    --hostname="127.0.0.1" \
+    --cert-dir="${CERT_DIR}" \
+    --latest-images \
+    --loglevel=${VERBOSE:-3} &> ${BASETMPDIR}/server.log &
+  echo -n $! > ${BASETMPDIR}/server.pid
+}
+
+start_docker_registry() {
+  mkdir -p ${BASETMPDIR}/.registry
+  openshift ex router --create --credentials="${KUBECONFIG}" \
+    --images='openshift/origin-${component}:latest'
+  openshift ex registry --create --credentials="${KUBECONFIG}" \
+    --mount-host="${BASETMPDIR}/.registry" \
+    --images='openshift/origin-${component}:latest'
+}
+
+# Go to the top of the tree.
+cd "${OS_ROOT}"
+
+# Start the Etcd server
+echo "[INFO] Starting etcd server (127.0.0.1:${ETCD_PORT})"
+start_etcd
+export ETCD_STARTED="1"
+
+# Start OpenShift sever that will be common for all extended test cases
+echo "[INFO] Starting OpenShift test server [${OS_EXTENDED_MASTER}]..."
+start_server
+trap cleanup EXIT SIGINT
+
+# Wait for the API server to come up
+wait_for_url_timed "https://${OS_EXTENDED_MASTER}/healthz" "" 90*TIME_SEC >/dev/null
+wait_for_url_timed "https://${OS_EXTENDED_MASTER}/osapi" "" 90*TIME_SEC >/dev/null
+wait_for_url "https://${OS_EXTENDED_MASTER}/api/v1beta1/minions/127.0.0.1" "" 0.25 80 >/dev/null
+
+# Start the Docker registry (172.30.17.101:5000)
+start_docker_registry
+
+wait_for_command '[[ "$(osc get endpoints docker-registry -t "{{ if .endpoints}}{{ len .endpoints }}{{ else }}0{{ end }}" 2>/dev/null || echo "0")" != "0" ]]' $((5*TIME_MIN))
+
+REGISTRY_ADDR=$(osc get --output-version=v1beta1 --template="{{ .portalIP }}:{{.port }}" \
+  service docker-registry)
+echo "[INFO] Verifying the docker-registry is up at ${REGISTRY_ADDR}"
+wait_for_url_timed "http://${REGISTRY_ADDR}" "" $((2*TIME_MIN))
+
+# Run all extended tests cases
+export OS_TEST_PACKAGE="test/extended"
+export OS_TEST_TAGS="extended"
+export OS_TEST_NAMESPACE="extended"
+
+[ ! -z "${RUN_SERVER-}" ] && wait;exit
```

Comment:
FYI, ignore this for now ;-) I'm using this to keep server running and don't run any test cases, so I can test helper methods/etc without restarting the server....